### PR TITLE
PMM-7358 manage components

### PR DESCRIPTION
**Pull request by BupycHuk**
[PMM-7358](https://jira.percona.com/browse/PMM-7358)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1653

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by palash25 — APPROVED** [review]
looks good, methods are missing comments though, would be nice if you could add them.

File: services/management/dbaas/components_service.go [hunk 1]
```diff
@@ -87,38 +96,113 @@ func (c componentsService) GetPXCComponents(ctx context.Context, req *dbaasv1bet
...
+	versions, err := c.versions(ctx, params, kubernetesCluster)
 	if err != nil {
 		return nil, err
 	}
 	return &dbaasv1beta1.GetPXCComponentsResponse{Versions: versions}, nil
 }
 
-func (c componentsService) versions(ctx context.Context, params componentsParams) ([]*dbaasv1beta1.Version, error) {
+func (c componentsService) ChangePSMDBComponents(ctx context.Context, req *dbaasv1beta1.ChangePSMDBComponentsRequest) (*dbaasv1beta1.ChangePSMDBComponentsResponse, error) {
+	err := c.db.InTransaction(func(tx *reform.TX) error {
```
Comment (jprukner) [inline services/management/dbaas/components_service.go:112]: InTransactionContext

File: services/management/dbaas/components_service.go [hunk 2]
```diff
@@ -87,38 +96,113 @@ func (c componentsService) GetPXCComponents(ctx context.Context, req *dbaasv1bet
...
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &dbaasv1beta1.ChangePSMDBComponentsResponse{}, nil
+}
+
+func (c componentsService) ChangePXCComponents(ctx context.Context, req *dbaasv1beta1.ChangePXCComponentsRequest) (*dbaasv1beta1.ChangePXCComponentsResponse, error) {
+	err := c.db.InTransaction(func(tx *reform.TX) error {
```
Comment (jprukner) [inline services/management/dbaas/components_service.go:141]: InTransactionContext

File: services/management/dbaas/components_service.go [hunk 3]
```diff
@@ -152,8 +236,54 @@ func (c componentsService) matrix(m map[string]component, minimalVersion *govers
...
+			return nil, fmt.Errorf("enable and disable for version %s can't be passed together", v.Version)
+		}
+		if v.Enable {
+			delete(disabledVersions, v.Version)
+		}
+		if v.Disable {
+			disabledVersions[v.Version] = struct{}{}
+		}
+	}
+	kc.DisabledVersions = make([]string, 0)
```
Comment (jprukner) [inline services/management/dbaas/components_service.go:281]:
```suggestion
	kc.DisabledVersions = make([]string, 0, len(disabledVersions))
```

**Review by jprukner — CHANGES_REQUESTED** [review]

File: services/management/dbaas/components_service.go (hunk 1 above)
Comment (BupycHuk) [reply services/management/dbaas/components_service.go:112]: I checked the code and we never used `InTransactionContext`, so if we will move to `InTransactionContext` we should do it everywhere for consistency

Comment (jprukner) [reply services/management/dbaas/components_service.go:113]: It's ok to skip it for now then. Resolving the conversation.

File: models/kubernetes_helpers_test.go [hunk 4]
```diff
@@ -53,6 +56,18 @@ func TestKubernetesHelpers(t *testing.T) {
...
+				PXC: &models.Component{
+					DisabledVersions: []string{"8.0.0"},
+					DefaultVersion:   "8.0.1-20",
+				},
+				ProxySQL: &models.Component{
+					DisabledVersions: []string{"8.0.0"},
+					DefaultVersion:   "8.0.1-19",
+				},
+				Mongod: &models.Component{
+					DisabledVersions: []string{"3.4.0", "3.6.0"},
```
Comment (jprukner) [inline models/kubernetes_helpers_test.go:68]: I am not sure these values have any effect. @BupycHuk

**Review by jprukner — APPROVED** [review]

File: models/kubernetes_helpers_test.go (hunk 4 above)
Comment (jprukner) [reply models/kubernetes_helpers_test.go:68]: Aside from that, it looks good to me.

Comment (BupycHuk) [reply models/kubernetes_helpers_test.go:68]: In this test we check only model layer, so values don't matter.

**Pull request merged**